truffle/src/main/java/org/jruby/truffle/nodes/core/BigDecimalNodes.java

```diff
@@ -194,34 +256,199 @@ public SubNode(RubyContext context, SourceSection sourceSection) {
+            return createRubyBigDecimal(
+                    getContext().getCoreLibrary().getBigDecimalClass(),
+                    getBigDecimalValue(a).multiply(b, new MathContext(precision)));
+        }
+
+        @Specialization
+        public Object mult(RubyBasicObject a, int b, int precision) {
+            return mulBigDecimal(a, getBigDecimalValue(b), precision);
+        }
+
+        @Specialization
+        public Object mult(RubyBasicObject a, long b, int precision) {
+            return mulBigDecimal(a, getBigDecimalValue(b), precision);
+        }
+
+        @Specialization
+        public Object mult(RubyBasicObject a, double b, int precision) {
+            return mulBigDecimal(a, getBigDecimalValue(b), precision);
+        }
+
+        @Specialization
+        public Object mult(RubyBasicObject a, RubyBignum b, int precision) {
+            return mulBigDecimal(a, getBigDecimalValue(b), precision);
+        }
+
+        @Specialization(guards = "isRubyBigDecimal(b)")
+        public Object mult(RubyBasicObject a, RubyBasicObject b, int precision) {
+            return mulBigDecimal(a, getBigDecimalValue(b), precision);
+        }
+
+    }
+
+    @CoreMethod(names = "/", required = 1)
+    public abstract static class DivOpNode extends BigDecimalCoreMethodNode {
+
+        public DivOpNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        private Object mulBigDecimal(RubyBasicObject a, BigDecimal b) {
+            // precision based on https://github.com/pitr-ch/jruby/blob/bigdecimal/core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java#L892-903
+            int len = getBigDecimalValue(a).precision() + b.precision();
+            int pow = len / 4;
+            int precision = (pow + 1) * 4 * 2;
+
+            return createRubyBigDecimal(
+                    getContext().getCoreLibrary().getBigDecimalClass(),
+                    getBigDecimalValue(a).divide(b, new MathContext(precision)));
+        }
+
+        @Specialization
+        public Object division(RubyBasicObject a, int b) {
+            return mulBigDecimal(a, getBigDecimalValue(b));
+        }
+
+        @Specialization
+        public Object division(RubyBasicObject a, long b) {
+            return mulBigDecimal(a, getBigDecimalValue(b));
+        }
+
+        @Specialization
+        public Object division(RubyBasicObject a, double b) {
+            return mulBigDecimal(a, getBigDecimalValue(b));
+        }
+
+        @Specialization
+        public Object division(RubyBasicObject a, RubyBignum b) {
+            return mulBigDecimal(a, getBigDecimalValue(b));
+        }
+
+        @Specialization(guards = "isRubyBigDecimal(b)")
+        public Object division(RubyBasicObject a, RubyBasicObject b) {
+            return mulBigDecimal(a, getBigDecimalValue(b));
+        }
+
+    }
+
+    // TODO div,quo,divmod
```

Comment:
Please name and date any TODOs. I know you can get the same info from blame but it's nice to scan and see if someone is a month old or a year old.